clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3581,11 +3648,13 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
 
 public:
   static constexpr const bool VisitInsideSimpleContainers = false;
-  SyclKernelIntHeaderCreator(Sema &S, SYCLIntegrationHeader &H,
+  SyclKernelIntHeaderCreator(KernelCallOperatorVisitor KernelCallOperator,
+                             Sema &S, SYCLIntegrationHeader &H,
                              const CXXRecordDecl *KernelObj, QualType NameType,
                              FunctionDecl *KernelFunc)
-      : SyclKernelFieldHandler(S), Header(H) {
-    bool IsSIMDKernel = isESIMDKernelType(KernelObj);
+      : SyclKernelFieldHandler(S), Header(H),
+        KernelCallOperator(KernelCallOperator) {
+    bool IsSIMDKernel = isESIMDKernelType(KernelCallOperator);
```

Comment:
BTW, it doesn't make sense to run traversal from each of visitors. For BodyCreator and others running from ConstructOpenCLKernel, you can also do traverse only one time.